### ISSUE TITLE
Content types with parameters validate wrong request data

### ISSUE THREAD

**Issue by jordandobrev** [cdimascio/express-openapi-validator#271]
When the same content type is used with different properties in the requestBody the validation always validates agains the first content type schema.

Example:
```
      requestBody:
        content:
          "application/vnd.company.user+json; type=basic":
            schema:
              $ref: '#/components/schemas/BasicUser'
         "application/vnd.company.user+json; type=internal":
            schema:
              $ref: '#/components/schemas/InternalUser'
```

Atm the content type validation passes but it always validates the request against `BasicUser` instead of `InternalUser` when the request is made with `application/vnd.company.user+json; type=internal`

**Comment by cdimascio** [comment]
This issue is similar to #270. The content matcher is ignoring the boundary. Perhaps, boundary should only be ignored for multipart?

**Comment by cdimascio** [comment]
fixed in `v3.12.0`
